### ISSUE TITLE
rtmidi_c.h struct issue

### ISSUE THREAD

**Issue by 8bitbubsy** [thestk/rtmidi#162]
rtmidi_c.h won't compile correctly on all C compilers (f.ex. Visual Studio) because of the RtMidiWrapper struct.
Here's a fix:

rtmidi_c.h before:
struct RtMidiWrapper {
[...]
};

rtmidi_c.h after (remember both RtMidiWrapper):
typedef struct RtMidiWrapper { 
[...]
} RtMidiWrapper; 

You should consider making a new release very soon as well. The C wrapper (when using callback) is useless in 3.00 because it lacks the size callback parameter, which was added in a later revision. This is a very critical change, without it you need to strlen() the char buffer which ends too early if the first data byte is zero.

**Comment by radarsat1** [comment]
What error does the Microsoft compiler give? Because that struct declaration appears fine to me.  Your proposed change re-uses the same identifier which seems unusual tbh.

Noted re the release.  I don't recommend strlen() on a MIDI message.

**Comment by 8bitbubsy** [comment]
Wow, that's really strange. I reopened the rtmidi project file, and now it suddenly compiles fine without the modifications at all. I guess this was a pointless bug report then. It was definitely complaining at some point, it didn't like the type when it was used in "typedef struct RtMidiWrapper* RtMidiPtr;" etc.
Oh well!

**Comment by radarsat1** [comment]
Hm, strange it should be ok to use structs like that.  I'll close for now but if you manage to reproduce it please re-open and post the error message.

**Comment by atsushieno** [comment]
Interestingly now I'm hitting this issue when I am trying to build zrythm (with `-Drtmidi=enabled`) with gcc (Ubuntu 9.3.0-10ubuntu2) 9.3.0:

```
In file included from ../inc/audio/ext_port.h:46,
                 from ../inc/audio/channel.h:34,
                 from ../src/gui/widgets/arranger.c:25:
/usr/include/rtmidi/rtmidi_c.h:31:9: error: unknown type name ‘RtMidiWrapper’
   31 | typedef RtMidiWrapper* RtMidiPtr;
      |         ^~~~~~~~~~~~~
/usr/include/rtmidi/rtmidi_c.h:34:9: error: unknown type name ‘RtMidiWrapper’
   34 | typedef RtMidiWrapper* RtMidiInPtr;
      |         ^~~~~~~~~~~~~
/usr/include/rtmidi/rtmidi_c.h:37:9: error: unknown type name ‘RtMidiWrapper’
   37 | typedef RtMidiWrapper* RtMidiOutPtr;
      |         ^~~~~~~~~~~~~
```

Making changes to them as `typedef struct ...` fixes the issue.

**Comment by atsushieno** [comment]
Nevermind, it seems already fixed at b261781.
